FileUploader: Add Customize FileUploader How To (#7488)
DevExpress/devextreme-documentation: concepts/05 UI Components/FileUploader/50 Customize FileUploader.md

This tutorial demonstrates how to customize FileUploader appearance.

<div class="simulator-desktop-container" data-view="/Content/Applications/25_1/UIWidgets/FileUploader/Customization/index.html, /Content/Applications/25_1/UIWidgets/FileUploader/Customization/index.js, /Content/Applications/25_1/UIWidgets/FileUploader/Customization/index.css"></div>

> Specify the [dropZone](/Documentation/ApiReference/UI_Components/dxFileUploader/Configuration/#dropZone) property to integrate a custom drop zone. Use the `width` and `height` CSS styles to modify the dimensions of the dropzone. This tutorial also implements the `background-color` and `border-radius` styles:

Reviewer: ```suggestion
Specify the [dropZone](/Documentation/ApiReference/UI_Components/dxFileUploader/Configuration/#dropZone) property to integrate a custom drop zone. Use the `width` and `height` CSS styles to modify drop zone dimensions. This tutorial also implements the `background-color` and `border-radius` styles:
```

        height: 200px;
        background-color: #f5f5f5;
        border-radius: 16px;
    }

##### Angular

    <!-- tab: app.component.html -->
    <div class="file-uploader-block"></div>
    <dx-file-uploader dropZone=".file-uploader-block"></dx-file-uploader>

    <!-- tab: app.component.css -->
    .file-uploader-block {
        width: 350px;
        height: 200px;
        background-color: #f5f5f5;
        border-radius: 16px;
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div class="file-uploader-block"></div>
        <DxFileUploader drop-zone=".file-uploader-block" />
    </template>

    <!-- tab: styles.css -->
    .file-uploader-block {
        width: 350px;
        height: 200px;
        background-color: #f5f5f5;
        border-radius: 16px;
    }

##### React

    <!-- tab: App.tsx -->
    export default function App() {
        return (
            <div class="file-uploader-block"></div>
            <FileUploader dropZone=".file-uploader-block" />
        )
    }

    <!-- tab: styles.css -->
    .file-uploader-block {
        width: 350px;
        height: 200px;
        background-color: #f5f5f5;
        border-radius: 16px;
    }

---

> This tutorial places FileUploader inside the custom drop zone container:

Reviewer: This is how I think about this sentence. 

We already applied some customizations to a FileUploader component (instructions and code above). So now it's "the FileUploader". 

But "custom drop zone container" is something new that we haven't worked with before. So I would refer to it as "a custom drop zone container" (this first time we mention it).

```suggestion
This tutorial places the FileUploader inside a custom drop zone container:
```

---

##### jQuery

    <!-- tab: index.html -->
    <div class="file-uploader-block">
        <div id="file-uploader"></div>
    </div>

    <!-- tab: index.js -->
    $(() => {
        $('#file-uploader').dxFileUploader({
            dropZone: '.file-uploader-block',
        })
    })

##### Angular

    <!-- tab: app.component.html -->
    <div class="file-uploader-block">
        <dx-file-uploader dropZone=".file-uploader-block"></dx-file-uploader>
    </div>

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div class="file-uploader-block">
            <DxFileUploader drop-zone=".file-uploader-block" />
        </div>
    </template>

##### React

    <!-- tab: App.tsx -->
    export default function App() {
        return (
            <div class="file-uploader-block">
                <FileUploader dropZone=".file-uploader-block" />
            </div>
        )
    }

---

> To center FileUploader elements inside a custom drop zone, assign the following CSS styles to the `.dx-fileuploader-input-wrapper` selector:

Reviewer: But now I assume we're still talking about the same custom drop zone. So I'd say "the drop zone". 

```suggestion
To center FileUploader elements inside the drop zone, assign the following CSS styles to the `.dx-fileuploader-input-wrapper` selector:
```

    <!-- tab: styles.css -->
    .dx-fileuploader-input-wrapper {
        position: relative;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
    }

> When users upload files, the component lists selected files in its container. To modify placement of this list, implement absolute positioning on the uploaded file container. Assign the following CSS styles to the `.dx-fileuploader-files-container` selector:

Reviewer: Can we say "move" instead of "modify placement"? 

```suggestion
When users upload files, the component lists selected files in its container. To move this list, specify absolute positioning. Assign the following CSS styles to the `.dx-fileuploader-files-container` selector:
```

Author: Yeah "move" is a much simpler way of conveying the same idea.

    <!-- tab: styles.css -->
    .dx-fileuploader-files-container {
        position: absolute;
        top: 70%;
        left: 50%;
        transform: translate(-50%, 0);
        width: 35%;
        height: 30%;
        overflow-y: auto;
    }

> In this tutorial, the uploaded file container is positioned below the FileUploader with the `top: 70%` style. The `overflow-y: auto` style adds a scroll bar to the uploaded file container when items overflow the element.

Reviewer: I usually try not to include pieces of code in the middle of the sentence, especially when those fragments are in the middle of the phrase - like "The `overflow-y: auto` style". It's just something you can't read comfortably. Since you list a couple of different styles here and explain their purpose, I would use a list. 

```suggestion
Note he following styles: 

- `top: 70%`: Positions the uploaded file list container below the FileUploader.
- `overflow-y: auto`: Adds a scroll bar to the uploaded file list container when necessary. 
```

Author: ```suggestion
Note the following styles: 

- `top: 70%`: Positions the uploaded file list container below the FileUploader.
- `overflow-y: auto`: Adds a scroll bar to the uploaded file list container when necessary.
```
